libxcfun "patch" issue in CMake

One should normally be able to "make" more than once in the build directory, yes? It should just immediately detect that the source code hasn't changed since the project was built and exit successfully, right?

I can compile PySCF from source using CMake without issue and it works, but then if for whatever reason I decide to try to `make` a second time, this happens:

```
(pyscf) herme068@herme068:build$ make
[  1%] Performing update step for 'libcint'
[  2%] No patch step for 'libcint'
[  3%] Performing configure step for 'libcint'
loading initial cache file /home/herme068/gits/pyscf/pyscf/lib/build/deps/tmp/libcint-cache-RELWITHDEBINFO.cmake
Build type RELEASE
Enabled WITH_F12
P orbitals convention (px, py, pz)
Minimal cutoff in exponent for integral screening 20
Do not trigger hard exit for numerical issues in Rys quadrature
Exclude Fortran interface
Exclude old cint (version 2) interface
-- Configuring done
-- Generating done
-- Build files have been written to: /home/herme068/gits/pyscf/pyscf/lib/build/deps/src/libcint-build
[  3%] Performing build step for 'libcint'
[100%] Built target cint
[  4%] Performing install step for 'libcint'
[100%] Built target cint
Install the project...
-- Install configuration: "RELEASE"
-- Up-to-date: /home/herme068/gits/pyscf/pyscf/lib/deps/lib/libcint.so.6.0.0
-- Up-to-date: /home/herme068/gits/pyscf/pyscf/lib/deps/lib/libcint.so.6
-- Up-to-date: /home/herme068/gits/pyscf/pyscf/lib/deps/lib/libcint.so
-- Up-to-date: /home/herme068/gits/pyscf/pyscf/lib/deps/include/cint_funcs.h
-- Up-to-date: /home/herme068/gits/pyscf/pyscf/lib/deps/include/cint.h
[  5%] Completed 'libcint'
[  7%] Built target libcint
[ 14%] Built target libxc
[ 15%] Performing update step for 'libxcfun'
[ 16%] Performing patch step for 'libxcfun'
error: patch failed: CMakeLists.txt:2
error: CMakeLists.txt: patch does not apply
error: patch failed: src/XCFunctional.cpp:432
error: src/XCFunctional.cpp: patch does not apply
make[2]: *** [CMakeFiles/libxcfun.dir/build.make:113: deps/src/libxcfun-stamp/libxcfun-patch] Error 1
make[1]: *** [CMakeFiles/Makefile2:324: CMakeFiles/libxcfun.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

--- a/pyscf/lib/CMakeLists.txt
+++ b/pyscf/lib/CMakeLists.txt
@@ -222,7 +222,7 @@ if(ENABLE_XCFUN AND BUILD_XCFUN)
     GIT_REPOSITORY https://github.com/dftlibs/xcfun.git
     GIT_TAG a89b783
     # This patch adds the xcfun derivative order up to 5
-    PATCH_COMMAND git apply ${PROJECT_SOURCE_DIR}/libxcfun.patch
+    PATCH_COMMAND git apply --reject ${PROJECT_SOURCE_DIR}/libxcfun.patch
     PREFIX ${PROJECT_BINARY_DIR}/deps
     INSTALL_DIR ${PROJECT_SOURCE_DIR}/deps
     CMAKE_ARGS -DCMAKE_BUILD_TYPE=RELEASE -DBUILD_SHARED_LIBS=1

